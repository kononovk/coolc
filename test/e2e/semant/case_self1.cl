class A {};
class B {};

class Main inherits IO {
    main(): Object {
      1
    };

    foo(obj: Object): Object {
      case obj of
        a: A => out_string("This is A.");
        b: B => out_string("This is B.");
        c: Object => out_string("Unknown case.");
      esac
    };
};
