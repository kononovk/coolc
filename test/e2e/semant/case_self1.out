#1
_program
  #1
  _class
    A
    Object
    "test/e2e/semant/case_self1.cl"
    (
    )
  #2
  _class
    B
    Object
    "test/e2e/semant/case_self1.cl"
    (
    )
  #4
  _class
    Main
    IO
    "test/e2e/semant/case_self1.cl"
    (
    #5
    _method
      main
      Object
      #6
      _int
        1
      : Int
    #9
    _method
      foo
      #9
      _formal
        obj
        Object
      Object
      #10
      _typcase
        #10
        _object
          obj
        : Object
        #11
        _branch
          a
          A
          #11
          _dispatch
            #11
            _object
              self
            : SELF_TYPE
            out_string
            (
            #11
            _string
              "This is A."
            : String
            )
          : SELF_TYPE
        #12
        _branch
          b
          B
          #12
          _dispatch
            #12
            _object
              self
            : SELF_TYPE
            out_string
            (
            #12
            _string
              "This is B."
            : String
            )
          : SELF_TYPE
        #13
        _branch
          c
          Object
          #13
          _dispatch
            #13
            _object
              self
            : SELF_TYPE
            out_string
            (
            #13
            _string
              "Unknown case."
            : String
            )
          : SELF_TYPE
      : SELF_TYPE
    )
